Add info on DynamoDB from Jira SPE-125 (#23682)
hashicorp/vault: website/content/docs/secrets/databases/index.mdx

disable_escaping="true"
```

## Unsupported databases

### AWS DynamoDB

Amazon Web Services (AWS) DynamoDB is a fully managed, serverless, key-value NoSQL database service. While
DynamoDB is not supported by the database secrets engine, you can use the [AWS secrets engine](/vault/docs/secrets/aws)
to provision dynamic credentials capable of accessing DynamoDB.

1. Verify you have the AWS secrets engine enabled and configured.

> 1. Create a role with the necessary permissions for your users to access DynamoDB.
> 
>    **Example:**

Reviewer: ```suggestion
1. Create a role with the necessary permissions for your users to access DynamoDB. For example:
```

   ```shell-session
   $ vault write aws/roles/aws-dynamodb-read \
           credential_type=iam_user \
           policy_document=-<<EOF
   {
   	"Version": "2012-10-17",
   	"Statement": [
   		{
   			"Effect": "Allow",
   			"Action": [
   				"dynamodb:DescribeTable",
   				"dynamodb:GetItem",
   				"dynamodb:GetRecords"
   			],
   			"Resource": "arn:aws:dynamodb:us-east-1:1234567891:table/example-table"
   		},
   		{
   			"Effect": "Allow",
   			"Action": "dynamodb:ListTables",
   			"Resource": "*"
   		}
   	]
   }
   EOF
   ```

> 1. Generate dynamic credentials for DynamoDB using the `aws-dynamodb-read` role.

Reviewer: ```suggestion
1. Generate dynamic credentials for DynamoDB using the `aws-dynamodb-read` role:
```

   ```shell-session
   $ vault read aws/creds/aws-dynamodb-read
   Key                Value
   ---                -----
   lease_id           aws/creds/my-role/kbSnl9WSDzOXQerd8GiVh75N.DACNl
   lease_duration     1h
   lease_renewable    true
   access_key         AKALMNOP123456
   secret_key         xY4XhS3AsM3s+R33tCaybsT2XI6BVL+vF+khbbYD
   security_token     <nil>
   ```

> 1. Use the dynamic credentials generated by Vault to access DynamoDB. For example, using

Reviewer: ```suggestion
1. Use the dynamic credentials generated by Vault to access DynamoDB. For example, to connect with the
```
